### PR TITLE
fix: restore 100% package coverage after Vitest 4 upgrade

### PR DESCRIPTION
**Pull request by ZappoMan**
## Summary
- Restores 100/100/100/100 coverage (statements, branches, functions, lines) across all Vitest packages after the Vitest 4.1.8 upgrade introduced stricter v8 branch counting.
- Adds targeted tests for async cancellation guards, dynamic Sentry import fallbacks, debounced realtime reloads, and other edge-case branches in billing, admin, connections, observability, waitlist, marketing-email, and shared-tests.
- Fixes a `BillingProvider` test mock leak (`getSession` hanging mock) and a minor `useUsage` ref typing issue uncovered by type-check.

## Test plan
- [x] `npm run test:coverage` — all package shards report 100% on statements, branches, functions, and lines
- [x] Pre-commit hooks pass (eslint, prettier, full monorepo type-check)
- [ ] CI unit-coverage-shard jobs pass on PR

### CODE REVIEW

**Review by mei-artificerinnovations — APPROVED** [review]
APPROVE — coverage restoration is clean and well-targeted.

Production change in `useUsage.ts` is correct: initializing the ref to `undefined` instead of a no-op function, with optional chaining on call, prevents the ref being invoked before it is assigned.

The `mockImplementation` → `mockImplementationOnce` fix in `BillingProvider.test.tsx` is a real bug fix — the hanging `getSession` mock was leaking state into subsequent tests in the suite.

All new tests are well-structured: the unmount-guard pattern is consistent across BillingProvider, useSignupMode, useConnections, and ObservabilityProvider. The `sentryDynamicImport.coverage.test.ts` correctly uses `vi.resetModules()` in `afterEach` to prevent module cache pollution between the dynamic-import tests. Everything else (pagination error path, Tab-trap guard, non-activation key guard, whitespace-only error string, `response.text()` rejection) tests real branches without over-mocking.

File: packages/connections/src/hooks/useConnections.test.tsx [hunk 1]
```diff
@@ -249,4 +249,44 @@ describe('useConnections', () => {
...
+    await result.current.refresh();
+    await waitFor(() => expect(result.current.rows).toHaveLength(0));
+    expect(calls).toBe(2);
+  });
+
+  it('ignores debounced reload after unmount', async () => {
+    vi.useFakeTimers();
+    const clearTimeoutSpy = vi
+      .spyOn(globalThis, 'clearTimeout')
+      .mockImplementation(() => undefined);
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply packages/connections/src/hooks/useConnections.test.tsx]: Fixed in c8710b2. Replaced the no-op clearTimeout stub with a call-through setTimeout spy that captures the 400ms debounce callback while delegating to the real implementation, plus a plain clearTimeout spy to verify cleanup. After unmount we manually invoke the captured callback to exercise the if (!cancelled) guard without masking timer cleanup behavior.

File: packages/connections/src/hooks/useConnections.test.tsx [hunk 2]
```diff
@@ -249,4 +249,44 @@ describe('useConnections', () => {
...
+    try {
+      let calls = 0;
+      const supabase = createMockSupabase(async () => {
+        calls += 1;
+        return { data: [row], error: null };
+      });
+      const { unmount } = renderHook(() =>
+        useConnections({ supabase, userId: UUID_A })
+      );
+      await vi.waitFor(() => expect(calls).toBe(1));
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply packages/connections/src/hooks/useConnections.test.tsx]: Fixed in c8710b2. Switched back to `waitFor` from `@testing-library/react`. This works correctly now that `setTimeout` delegates to the original implementation instead of being fully replaced, so React updates are wrapped in `act` consistently with the rest of the file.

File: packages/observability/src/__tests__/ObservabilityProvider.web.test.tsx [hunk 3]
```diff
@@ -142,4 +142,17 @@ describe('ObservabilityProvider (web)', () => {
...
+      <ObservabilityProvider config={config}>
+        <div>child</div>
+      </ObservabilityProvider>
+    );
+    unmount();
+    await act(async () => {
+      await Promise.resolve();
+    });
+    expect(SentryMock.setUser).not.toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply packages/observability/src/__tests__/ObservabilityProvider.web.test.tsx:180]: Fixed in c8710b2. Rewrote the test to use `vi.resetModules` with a deferred `@sentry/react` dynamic import: unmount the provider, then resolve the pending Sentry load, and assert no `console.error` warnings about updating an unmounted component. This directly validates the cancellation guard on `setSentry` rather than checking `setUser`, which was never invoked on this codepath.

**Review by mei-artificerinnovations — APPROVED** [review]
APPROVE — both Copilot fixes are genuine improvements.

The useConnections debounce test now uses a call-through setTimeout spy that captures the callback at delay=400, fires it manually after unmount, and asserts clearTimeout was called. Strictly better than the previous fake-timer + stubbed clearTimeout approach.

The ObservabilityProvider unmount test now actually exercises the race: vi.resetModules() + vi.doMock with a manually-controlled pending Promise means Sentry resolves after unmount for real, and the console.error check for unmounted warnings is the right signal. The previous version tested with Sentry already resolved, so the guard was never under pressure. Both fixes correct.
